Tests/Optiq.DatePicker.Tests/DatePicker/DatePickerUnitTests.cs

```diff
@@ -0,0 +1,210 @@
+        picker.MaximumDate = newDate;
+
+        Assert.True(maximumDateChanged);
+        Assert.True(dateChanged);
+
+        Assert.Equal(newDate, picker.MaximumDate);
+        Assert.Equal(newDate, picker.Date);
+        Assert.Equal(picker.MaximumDate, picker.Date);
+    }
+
+    [Fact]
+    public void TestMinimumDateClamping()
+    {
+        Optiq.DatePicker.DatePicker picker = new Optiq.DatePicker.DatePicker();
+
+        picker.Date = new DateTime(1950, 1, 1);
+
+        Assert.Equal(new DateTime(1950, 1, 1), picker.Date);
+
+        bool dateChanged = false;
+        bool minimumDateChanged = false;
+        picker.PropertyChanged += (sender, e) =>
+        {
+            switch (e.PropertyName)
+            {
+                case "MinimumDate":
+                    minimumDateChanged = true;
+                    break;
+                case "Date":
+                    dateChanged = true;
+                    Assert.False(minimumDateChanged);
+                    break;
+            }
+        };
+
+        var newDate = new DateTime(2000, 1, 1);
+        picker.MinimumDate = newDate;
+
+        Assert.True(minimumDateChanged);
+        Assert.True(dateChanged);
+
+        Assert.Equal(newDate, picker.MinimumDate);
+        Assert.Equal(newDate, picker.Date);
+        Assert.Equal(picker.MinimumDate, picker.Date);
+    }
+
+    [Fact]
+    public void TestDateClamping()
+    {
+        Optiq.DatePicker.DatePicker picker = new Optiq.DatePicker.DatePicker();
+
+        picker.Date = new DateTime(1500, 1, 1);
+
+        Assert.Equal(picker.MinimumDate, picker.Date);
+
+        picker.Date = new DateTime(2500, 1, 1);
+
+        Assert.Equal(picker.MaximumDate, picker.Date);
+    }
+
+    [Fact]
+    public void TestDateSelected()
+    {
+        Optiq.DatePicker.DatePicker picker = new Optiq.DatePicker.DatePicker();
+
+        bool selected = false;
+        picker.DateSelected += (sender, arg) => selected = true;
+
+        // we can be fairly sure it wont ever be 2008 again
+        picker.Date = new DateTime(2008, 5, 5);
+
+        Assert.True(selected);
+    }
+
+    public static object[] DateTimes = {
+            new object[] { new DateTime (2006, 12, 20), new DateTime (2011, 11, 30) },
+            new object[] { new DateTime (1900, 1, 1), new DateTime (1999, 01, 15) }, // Minimum Date
+			new object[] { new DateTime (2006, 12, 20), new DateTime (2100, 12, 31) } // Maximum Date
+		};
+
+    public static IEnumerable<object[]> DateTimesData()
+    {
+        foreach (var o in DateTimes)
+        {
+            yield return o as object[];
+        }
+    }
+
+    [Theory, MemberData(nameof(DateTimesData))]
+    public void DatePickerSelectedEventArgs(DateTime initialDate, DateTime finalDate)
+    {
+        var datePicker = new Optiq.DatePicker.DatePicker();
+        datePicker.Date = initialDate;
+
+        Optiq.DatePicker.DatePicker pickerFromSender = null;
+        DateTime? oldDate = new DateTime();
```

Comment:
why don't you initialize those variables with `null`? `new DateTime()` looks very strange to me